torchdynamo/config.py

```diff
@@ -22,12 +22,69 @@
 # WARN print warnings (including graph breaks)
 # ERROR print exceptions (and what user code was being processed when it occurred)
 log_level = logging.WARNING
-# Verbose will print full stack traces on warnings and errors
-verbose = False
+
+
+def _get_loggers():
+    return [
+        logging.getLogger("torchdynamo"),
+        logging.getLogger("torchinductor"),
+    ]
+
+
+def _set_loggers_level(level):
+    for logger in _get_loggers():
+        logger.setLevel(level)
+
+
+_set_loggers_level(log_level)
 
 # the name of a file to write the logs to
 log_file_name = None
 
+LOGGING_CONFIG = {
+    "version": 1,
+    "formatters": {
+        "torchdynamo_format": {"format": "%(name)s: [%(levelname)s] %(message)s"},
+    },
+    "handlers": {
+        "torchdynamo_console": {
+            "class": "logging.StreamHandler",
+            "level": "DEBUG",
+            "formatter": "torchdynamo_format",
+            "stream": "ext://sys.stdout",
+        },
+    },
+    "loggers": {
+        "torchdynamo": {
+            "level": "DEBUG",
+            "handlers": ["torchdynamo_console"],
+            "propagate": False,
+        },
+        "torchinductor": {
+            "level": "DEBUG",
+            "handlers": ["torchdynamo_console"],
+            "propagate": False,
+        },
+    },
+    "disable_existing_loggers": False,
+}
+
+
+def init_logging():
+    if "PYTEST_CURRENT_TEST" not in os.environ:
+        logging.config.dictConfig(LOGGING_CONFIG)
+        # previous dictConfig call may overwrite logger levels
+        _set_loggers_level(log_level)
+        if log_file_name is not None:
+            log_file = logging.FileHandler(log_file_name)
+            log_file.setLevel(log_level)
+            for logger in _get_loggers():
+                logger.addHandler(log_file)
```

Comment:
Is it possible to leave init_logging and the original dict config inside utils.py and import it here? The main purpose of config.py is to contain config assignments, this feels out of place to me.

Reply:
Moved back to utils.